deepinv/training/trainer.py

```diff
@@ -566,6 +644,9 @@ def compute_metrics(
                     x_net=x_net,
                     x=x,
                     epoch=epoch,
+                    model=self.model,
```

Comment:
Why have you added these three lines? I remember us discussing this when I created the `Metric` interface, and you said that we shouldn't allow metrics to take `y`, `model` or `physics`

In fact, this is why I had to caveat QNR, if you wanted to use QNR in trainer you would've had to override `compute_metrics`
https://deepinv.github.io/deepinv/api/stubs/deepinv.loss.metric.QNR.html#deepinv.loss.metric.QNR.metric